src/cpp/src/visual_language/inputs_embedder.cpp

```diff
@@ -183,18 +191,25 @@ class InputsEmbedder::IInputsEmbedder {
                 encoded_input_ids = utils::subtract_chat_tokenized_inputs(
                     {new_chat_tokens}, prev_chat_tokens
                 ).input_ids;
+
+                if (m_last_disappeared_token.has_value() && *m_last_disappeared_token == encoded_input_ids.data<int64_t>()[0]) {
+                    m_last_disappeared_token = std::nullopt;
+                }
+
+                encoded_input_ids = ov::genai::utils::push_front_inputs(encoded_input_ids, std::vector<int64_t>{*m_last_disappeared_token});
             }
             auto end_tokenizer_time = std::chrono::steady_clock::now();
             metrics.raw_metrics.tokenization_durations.emplace_back(PerfMetrics::get_microsec(end_tokenizer_time - start_tokenizer_time));
             m_templated_chat_history = std::move(new_templated_chat_history);
-            m_tokenized_chat_history.clear();
-            std::copy(new_chat_tokens.data<int64_t>(), new_chat_tokens.data<int64_t>() + new_chat_tokens.get_size(),
-                        std::back_inserter(m_tokenized_chat_history));
+            m_tokenized_history.clear();
+            std::copy_n(new_chat_tokens.data<int64_t>(), new_chat_tokens.get_size(), std::back_inserter(m_tokenized_history));
         } else {
             auto start_tokenizer_time = std::chrono::steady_clock::now();
             encoded_input_ids = m_tokenizer.encode(prompt).input_ids;
             auto end_tokenizer_time = std::chrono::steady_clock::now();
             metrics.raw_metrics.tokenization_durations.emplace_back(PerfMetrics::get_microsec(end_tokenizer_time - start_tokenizer_time));
+            m_tokenized_history.clear();
```

Comment:
if we want to put real prompt to sampler, `m_tokenized_history` could be used. In `generate` we have prompt as string and we have `inputs_embeds`, which contains processed float values, but we don't have tokenized prompt and we can get it from `m_tokenized_history` if use it for non-chat too. Also for non-chat mode I can encode prompt explicitly in `generate`, but I thought it would be better to encode it once. Which way do you think is better?

Reply:
ok, now I understand the idea. And the idea of renaming `m_tokenized_chat_history` => `m_tokenized_history`

Looks good, but let's rename in all places to be consistent.

Reply:
renamed